pkg/util/log/log.go

```diff
@@ -105,6 +106,16 @@ func EnrichWithResourceID(log *logrus.Entry, resourceID string) *logrus.Entry {
 	})
 }
 
+// EnrichWithErrorStackTrace sets stacktrace log field if the error has a stacktrace
+func EnrichWithErrorStackTrace(log *logrus.Entry, err error) *logrus.Entry {
+	if err, ok := err.(interface {
+		StackTrace() errors.StackTrace
+	}); ok {
+		return log.WithField("stacktrace", fmt.Sprintf("%v", err.StackTrace()))
```

Comment:
I prefer `%v` over `%+v` - in our case this will be sufficient and will add only minimal "noise" to the logs. The example below shows the difference between the two (first one is `%v`)
```
{
  "msg": "parsing failed for . Invalid resource Id format",
  "component": "backend",
  "stacktrace": "[create.go:136 openshiftcluster.go:97 openshiftcluster.go:70 asm_amd64.s:1357]"
}
{
  "msg": "parsing failed for . Invalid resource Id format",
  "component": "backend",
  "stacktrace": "\ngithub.com/Azure/ARO-RP/pkg/backend/openshiftcluster.(*Manager).Create\n\t/home/omirensk/go/src/github.com/Azure/ARO-RP/pkg/backend/openshiftcluster/create.go:136\ngithub.com/Azure/ARO-RP/pkg/backend.(*openShiftClusterBackend).handle\n\t/home/omirensk/go/src/github.com/Azure/ARO-RP/pkg/backend/openshiftcluster.go:97\ngithub.com/Azure/ARO-RP/pkg/backend.(*openShiftClusterBackend).try.func1\n\t/home/omirensk/go/src/github.com/Azure/ARO-RP/pkg/backend/openshiftcluster.go:70\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1357"
}
``` 
using: ` journalctl _COMM=aro -o json --since "10 min ago"  | jq '{msg: .MESSAGE, component: .COMPONENT, stacktrace: .STACKTRACE}'`

Reply:
I wonder if `%+v` looks less bad in Geneva.
If we did this I think we'd probably remove `logrus.SetReportCaller(true)` above.

Reply:
Do you mean we remove `SetReportCaller` completely (delete https://github.com/Azure/ARO-RP/blob/cd03935243089d6c703e6365bce02b7ceabf6cab/pkg/util/log/log.go#L32) or just setting this to false when printing errors with stacktrace?
I don't think we can remove this completely because in this case we will lose this information for all the logs. 
But even in the case of the errors - we would still lose the place where the error itself is logged, because this information is not part of the error stacktrace. 

In this example these fields will not be available on the error:
```
"FUNCTION": "backend.(*openShiftClusterBackend).updateAsyncOperation.func1()",
"FILE": "pkg/backend/openshiftcluster.go:197",
```
this is the line where the error is printed

Reply:
I don't think we should remove reportcaller as what @olga-mirensky is adding is just for errors and info/warn/debug